pkg/storegateway/bucket.go

```diff
@@ -1635,49 +1577,29 @@ func (s *bucketBlockSet) remove(id ulid.ULID) {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
 
-	for i, bs := range s.blocks {
-		for j, b := range bs {
-			if b.meta.ULID != id {
-				continue
-			}
-			s.blocks[i] = append(bs[:j], bs[j+1:]...)
-			return
-		}
-	}
-}
-
-func int64index(s []int64, x int64) int {
-	for i, v := range s {
-		if v == x {
-			return i
+	for i, b := range s.blocks {
+		if b.meta.ULID != id {
+			continue
 		}
+		s.blocks = append(s.blocks[:i], s.blocks[i+1:]...)
+		return
 	}
-	return -1
 }
 
 // getFor returns a time-ordered list of blocks that cover date between mint and maxt.
-// Blocks with the biggest resolution possible but not bigger than the given max resolution are returned.
 // It supports overlapping blocks.
 //
 // NOTE: s.blocks are expected to be sorted in minTime order.
-func (s *bucketBlockSet) getFor(mint, maxt, maxResolutionMillis int64, blockMatchers []*labels.Matcher) (bs []*bucketBlock) {
+func (s *bucketBlockSet) getFor(mint, maxt int64, blockMatchers []*labels.Matcher) (bs []*bucketBlock) {
```

Comment:
Note to reviewers: the new logic is to select all blocks matching the input `mint`, `maxt` and `blockMatchers`. That's all. We don't need any recursion or complicated logic as before because we don't support resolutions.